test/integration/zk-discovery.bats

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bats
+
+load helpers
+
+ZK_HOST=127.0.0.1:2181
+
+function start_zk() {
+	docker run --name zk -p $ZK_HOST:2181 -d jplock/zookeeper
+}
+
+function stop_zk() {
+	docker rm -f -v zk > /dev/null
+}
+
+function teardown() {
+	stop_swarm_agents
+	stop_manager
+	stop_docker
+	stop_zk
+}
+
+@test "zk discovery should be working correctly" {
+	start_zk
+	start_docker 3
```

Comment:
I think just having one or two engines should prove this works, no need for so many (will speed up the test)